plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/marker/resolution/AbstractFBErrorMarkerResolution.java

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Primetals Technologies Austria GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Sebastian Hollersbacher
+ *    - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.application.marker.resolution;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.fordiac.ide.model.errormarker.FordiacErrorMarker;
+import org.eclipse.fordiac.ide.model.libraryElement.ErrorMarkerFBNElement;
+import org.eclipse.fordiac.ide.model.libraryElement.FB;
+import org.eclipse.fordiac.ide.model.libraryElement.FBType;
+import org.eclipse.fordiac.ide.model.libraryElement.LibraryElement;
+import org.eclipse.fordiac.ide.model.typelibrary.TypeLibrary;
+import org.eclipse.fordiac.ide.model.typelibrary.impl.ErrorFBTypeEntryImpl;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.ui.views.markers.WorkbenchMarkerResolution;
+
+public abstract class AbstractFBErrorMarkerResolution extends WorkbenchMarkerResolution {
+
+	private final String label;
+	private final IMarker marker;
+
+	protected AbstractFBErrorMarkerResolution(final String label, final IMarker marker) {
+		this.label = label;
+		this.marker = marker;
+	}
+
+	public TypeLibrary getTypeLibrary() {
+		try {
+			final EObject target = FordiacErrorMarker.getTarget(marker);
+			if (EcoreUtil.getRootContainer(target) instanceof final LibraryElement le) {
+				return le.getTypeLibrary();
+			}
+		} catch (final IllegalArgumentException | CoreException e) {
+			// target not found
+		}
+		return null;
```

Comment:
I would suggest getting the type library via the resource of the marker, which avoids having to potentially load the target object:
```suggestion
		return TypeLibraryManager.INSTANCE.getTypeLibrary(marker.getResource().getProject());
```